### ISSUE TITLE
[Bug]: Keyboard control for popover menu items is not working

### ISSUE THREAD

**Issue by Spark450** [GovAlta/ui-components#2238]
### Describe the bug

In the Navigation header example with Goa App Header items can be shown but are not reachable or navigatable using the keyboard. 

[See dev Jam](https://jam.dev/c/dbd3b354-a3e5-4fad-97e3-a438ff1417a3)

### Provide Code

Example created using example code on DS website

### To Reproduce

Go to "[Navigation header](https://design.alberta.ca/components/header#tab-0)" component and navigate page using tab key. When "tickets" menu is open try to navigate items in the popover using the keyboard.

**Comment by lattaai1** [comment]
Hello, I've fixed this issue!

This is what I've done:
I've made changes to improve keyboard navigation in the AppHeaderMenu component. The main issue was that the menu items were not reachable using the keyboard when the menu was open. Here's what I did:

1. In the AppHeaderMenu component, I added keyboard event handling to allow navigation through menu items using arrow keys and selection using Enter or Space.

2. I also ensured that when the menu is opened, focus is moved to the first menu item, and when closed, focus is returned to the menu button.

3. In the AppHeader component, I made sure that the menu toggle button is properly accessible and can be activated using the keyboard.

Here are the key changes in the AppHeaderMenu.svelte file:

I also created pull request: https://github.com/GovAlta/ui-components/pull/2240


> [!CAUTION]  
> Disclaimer: This fix was created by Latta AI and you should never merge before you check the correctness of generated code!

---
This bug was fixed for free by Latta AI - https://latta.ai/ourmission
